src/sourmash/index/__init__.py

```diff
@@ -49,6 +49,13 @@
                               flatten_and_downsample_num,
                               flatten_and_intersect_scaled)
 
+def _check_for_unknown_select_params(kwargs):
+    legit_param_names = { 'ksize', 'moltype', 'scaled', 'num', 'abund',
+                          'containment', 'picklist', 'filter_manifest' } # @CTB picklist
+    kk = set(kwargs.keys())
+    assert kk.issubset(legit_param_names), kk - legit_param_names
+
+
```

Comment:
Might be relevant: in #2726 I defined a [Selection TypedDict](https://github.com/sourmash-bio/sourmash/pull/2726/files#diff-8e86f05f9adfe96d7e26b80a68f3a3304b3ac819a346b3cfb231e6a5104515ce) to be able to
1) use type checking
2) being able to more easily convert a selection to Rust (the `_selection_as_rust` function).

Then the method signature for `select` becomes
`def select(self, **kwargs: Unpack[Selection]):`

(This is slightly annoying because the `select` method needs to extract the values from `kwargs`...)

[PEP-692](https://peps.python.org/pep-0692/) has more details